src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -0,0 +1,284 @@
+
+            if (propertyArray != null)
+            {
+                ArrayPool<byte>.Shared.Return(propertyArray);
+            }
+        }
+
+        private void WriteStringByOptions(ReadOnlySpan<char> propertyName, DateTime value)
+        {
+            ValidateWritingProperty();
+            if (_writerOptions.Indented)
+            {
+                WriteStringIndented(propertyName, value);
+            }
+            else
+            {
+                WriteStringMinimized(propertyName, value);
+            }
+        }
+
+        private void WriteStringByOptions(ReadOnlySpan<byte> propertyName, DateTime value)
+        {
+            ValidateWritingProperty();
+            if (_writerOptions.Indented)
+            {
+                WriteStringIndented(propertyName, value);
+            }
+            else
+            {
+                WriteStringMinimized(propertyName, value);
+            }
+        }
+
+        private void WriteStringMinimized(ReadOnlySpan<char> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameMinimized(escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringMinimized(ReadOnlySpan<byte> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameMinimized(escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringIndented(ReadOnlySpan<char> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameIndented(escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringIndented(ReadOnlySpan<byte> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameIndented(escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringValue(DateTime value, ref int idx)
+        {
+            if (_buffer.Length <= idx)
+            {
+                AdvanceAndGrow(ref idx);
+            }
+            _buffer[idx++] = JsonConstants.Quote;
+
+            FormatLoop(value, ref idx);
+
+            if (_buffer.Length <= idx)
+            {
+                AdvanceAndGrow(ref idx);
+            }
+            _buffer[idx++] = JsonConstants.Quote;
+        }
+
+        private static readonly StandardFormat s_dateTimeStandardFormat = new StandardFormat('O');
+
+        private void FormatLoop(DateTime value, ref int idx)
+        {
+            if (!Utf8Formatter.TryFormat(value, _buffer.Slice(idx), out int bytesWritten, s_dateTimeStandardFormat))
+            {
+                AdvanceAndGrow(ref idx, JsonConstants.MaximumFormatDateTimeLength);
+                Utf8Formatter.TryFormat(value, _buffer, out bytesWritten, s_dateTimeStandardFormat);
```

Comment:
I really wish that `Utf8Formatter` had followed the API guidance for Try-Parse of having the non-Try variants; because that's what should be called here.

I'll leave it to you to decide how you want to handle this.  In security code I always check the result of the Try APIs, and the ones that "can't" fail Debug.Fail and throw an unhelpful exception in their false case.  A less invasive option is to save the return value to a local and Debug.Assert that it's true... it at least helps track down where something went wrong if it ever does.